lib/ansible/module_utils/network/nxos/nxos.py

```diff
@@ -148,18 +148,42 @@ def get_config(self, flags=None):
             self._device_configs[cmd] = cfg
             return cfg
 
-    def run_commands(self, commands, check_rc=True, return_error=False):
+    def run_commands(self, commands, check_rc=True):
         """Run list of commands on remote device and return results
         """
         connection = self._get_connection()
 
-        if return_error:
+        try:
             return connection.run_commands(commands, check_rc)
-        else:
-            try:
-                return connection.run_commands(commands, check_rc)
-            except ConnectionError as exc:
-                self._module.fail_json(msg=to_text(exc))
+        except ConnectionError as exc:
+            self._module.fail_json(msg=to_text(exc))
+
+    def cli_run_commands(self, commands, check_rc=True):
+        """
+        Sends text command if JSON command fails for cliconf
+        when check_rc is set to False
+        """
+
+        if check_rc:
+            return self.run_commands(commands, check_rc)
+
+        if not check_rc:
+            out = self.run_commands(commands, check_rc)
+            capabilities = self.get_capabilities()
+            network_api = capabilities.get('network_api')
+            if network_api == 'cliconf':
+                if 'Invalid command at' in out and 'json' in out:
+                    for item in commands:
+                        if item['output'] == 'json':
```

Comment:
Check from line 177-186 is taken care of in cliconf run_command api. 
I think these lines can be replaced with `item['output'] = 'text'`